app/src/main/java/com/github/mik629/android/fundamentals/data/MoviesLoader.kt

```diff
@@ -0,0 +1,51 @@
+package com.github.mik629.android.fundamentals.data
+
+import com.github.mik629.android.fundamentals.data.db.daos.MovieDao
+import com.github.mik629.android.fundamentals.data.db.models.MovieWithActorsAndGenres
+import com.github.mik629.android.fundamentals.data.db.models.toMovie
+import com.github.mik629.android.fundamentals.data.db.models.toMovieDetails
+import com.github.mik629.android.fundamentals.data.network.ServerApi
+import com.github.mik629.android.fundamentals.data.network.model.ActorDTO
+import com.github.mik629.android.fundamentals.data.network.model.toActor
+import com.github.mik629.android.fundamentals.data.network.model.toMovie
+import com.github.mik629.android.fundamentals.data.network.model.toMovieDetails
+import com.github.mik629.android.fundamentals.domain.model.Movie
+import com.github.mik629.android.fundamentals.domain.model.MovieDetails
+import timber.log.Timber
+
+class MoviesLoader(
+    private val serverApi: ServerApi,
+    private val dao: MovieDao
+) {
+    suspend fun loadMoviesFromCache() =
+        dao.getAllMovies()
+            .map(MovieWithActorsAndGenres::toMovie)
+
+    suspend fun loadMoviesFromNetwork(category: String = "popular"): List<Movie> {
+        Timber.d("Loading from the network")
+        return serverApi.getMovieList(category)
+            .results
+            .map { movie ->
+                val movieDetails = serverApi.getMovieDetails(movie.id)
```

Comment:
can be loaded in parallel

Reply:
?

Reply:
I just don't understand how  am I going to set the rating - each time submit a new list of `Star` objects ordered by index having different values for a flag `isGrey` 🤔

Reply:
Also created an [issue](https://github.com/mik629/AndroidFundamentalsApp/issues/9)